src/persistence/SubjectExecutor.ts

```diff
@@ -654,15 +655,18 @@ export class SubjectExecutor {
         }
 
         // Run nested set updates one by one
-        const nestedSetPromise = new Promise<void>(async (ok, fail) => {
-            for (const subject of nestedSetSubjects) {
-                try {
-                    await updateSubject(subject)
-                } catch (error) {
-                    fail(error)
+        const nestedSetPromise = new Promise<void>((ok, fail) => {
+            (async () => {
+                for (const subject of nestedSetSubjects) {
+                    try {
+                        await updateSubject(subject)
+                    } catch (error) {
+                        fail(error)
+                        return
+                    }
                 }
-            }
-            ok()
+                ok()
+            })()
```

Comment:
There's no need to make this code more complex than it already is.

First, I'd get rid of the `return` that was added, it is unnecessary if the try-catch would not be inside the for loop.

Second, every async function returns a promise, so no need to use the Promise constructor. Unfortunately, this means that we have to have an IIFE like in this change, but still, it's more readable:

```ts
const nestedSetPromise = (async () => {
    for (const subject of nestedSetSubjects) {
        await updateSubject(subject)
    }
})()
```